app/src/main/java/com/cp/tasky/ui/sharecomponent/ClickableText.kt

```diff
@@ -0,0 +1,77 @@
+package com.cp.tasky.ui.sharecomponent
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.text.ClickableText
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.SpanStyle
+import androidx.compose.ui.text.buildAnnotatedString
+import androidx.compose.ui.text.toUpperCase
+import androidx.compose.ui.text.withStyle
+import androidx.compose.ui.tooling.preview.Preview
+
+@Composable
+fun PostClickableText(
+    modifier: Modifier = Modifier,
+    text: String,
+    clickableText: String,
+    onTextClick: () -> Unit
+) {
+    val nonClickableString by remember {
+        mutableStateOf(
+            text.substring(
+                0,
+                text.indexOf(clickableText)
+            )
+        )
+    }
```

Comment:
If you make this property a `val`, you say that it won't change. Why would you need to make it a state then, if it can't change?

Reply:
you are correct, this nonClickableString will be never changed. it should not be state at all. What do you recommend to change this. 

I am thinking if doing this way: `    val nonClickableString = text.substring(0, text.indexOf(clickableText))` , this will compile every single time when the compose is recomposed. Somehow I need that `remember`